cli/create.go

```diff
@@ -249,6 +248,17 @@ func createSandbox(ociSpec oci.CompatOCISpec, runtimeConfig oci.RuntimeConfig,
 func createContainer(ociSpec oci.CompatOCISpec, containerID, bundlePath,
 	console string, disableOutput bool) (vc.Process, error) {
 
+	// Set the mount type to 'ephemeral' if the mount source path
+	// is provisioned by k8s for ephemeral storage.
+	// For the given pod ephemeral volume is created only once
+	// backed by tmpfs inside the VM. For successive containers
+	// of the same pod the already existing volume is reused.
+	for idx, mnt := range ociSpec.Mounts {
+		if IsEphemeralStorage(mnt.Source) {
+			ociSpec.Mounts[idx].Type = "ephemeral"
```

Comment:
I think we need to define the type `ephemeral` (as other types) inside a separate package that can be imported both from the agent and the runtime. This can be done in a follow up PR, but this would be better than simply hardcode the type at each end of the chain (runtime on one side, and the agent on the other side.)